cmd/buildtools/helm.go

```diff
@@ -188,3 +195,59 @@ func (h *Helm) Push(path, dst string) error {
 	}
 	return up.UploadTo(path, dst)
 }
+
+func (h *Helm) Render(chartName string, chartPath string, vals map[string]interface{}, namespace string) ([]string, error) {
+
+	const k8sVersionMajor, k8sVersionMinor = "1", "26"
+
+	cfg := &action.Configuration{}
+
+	client := action.NewInstall(cfg)
+
+	client.DryRun = true
+	client.ReleaseName = chartName
+	client.Replace = true
+	client.ClientOnly = true
+	client.IncludeCRDs = true
+	client.Namespace = namespace
+
+	// Since ClientOnly is true we need to
+	// initialize KubeVersion otherwise resorts
+	// defaults
+	client.KubeVersion = &chartutil.KubeVersion{
+		Version: fmt.Sprintf("v%s.%s.0", k8sVersionMajor, k8sVersionMinor),
+		Major:   k8sVersionMajor,
+		Minor:   k8sVersionMinor,
+	}
+
+	chartRequested, err := loader.Load(chartPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load chart")
+	}
+
+	if req := chartRequested.Metadata.Dependencies; req != nil {
+		if err := action.CheckDependencies(chartRequested, req); err != nil {
+			return nil, errors.Wrap(err, "failed dependency check")
+		}
+	}
+
+	rel, err := client.Run(chartRequested, vals)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to render chart")
+	}
+
+	var manifests bytes.Buffer
+	fmt.Fprintln(&manifests, strings.TrimSpace(rel.Manifest))
+	for _, m := range rel.Hooks {
+		fmt.Fprintf(&manifests, "---\n# Source: %s\n%s\n", m.Path, m.Manifest)
+	}
```

Comment:
IIUC we are sorta removing all spaces from `rel.Manifest` but we don't do the same with `m.Manifest`. Maybe the why will become clear as I keep reading the code.